hal/include/HAL/Relay.h

```diff
@@ -9,11 +9,15 @@
 
 #include <stdint.h>
 
+#include "Handles.h"
+
 extern "C" {
-bool checkRelayChannel(void* digital_port_pointer);
+HalRelayHandle initializeRelayPort(HalPortHandle port_handle, uint8_t fwd,
+                                   int32_t* status);
+void freeRelayPort(HalRelayHandle relay_port_handle);
+
+bool checkRelayChannel(uint8_t pin);
 
-void setRelayForward(void* digital_port_pointer, bool on, int32_t* status);
-void setRelayReverse(void* digital_port_pointer, bool on, int32_t* status);
-bool getRelayForward(void* digital_port_pointer, int32_t* status);
-bool getRelayReverse(void* digital_port_pointer, int32_t* status);
+void setRelay(HalRelayHandle relay_port_handle, bool on, int32_t* status);
```

Comment:
That was the only way I could really do it. The WPILib had 8 resource counts, that way you could allocate separate objects for doing forward and reverse. I tried a few methods to get that working at the HAL level, and this was by far the cleanest and most forward compatible way I could do so. 

So now the HAL has 8 resources (1 for each pin). 0-3 are the forward pins, and when reverse is detected it gets allocated resources 4-7. Then the handle structure has a bool in it which detects forward or revers and it has the port number 0-3. When are then used in the get and set functions to switch between forward and reverse.